src/main/java/duke/ui/Ui.java

```diff
@@ -0,0 +1,160 @@
+
+import java.io.PrintStream;
+import java.util.Scanner;
+
+import duke.task.Task;
+import duke.tasklist.TaskList;
+
+
+
+/**
+ * Ui class responsible for handling Ui and error messages
+ */
+public class Ui {
+
+    private Scanner sc;
+    private PrintStream out;
+
+    /**
+     * Constructor of Ui
+     */
+    public Ui() {
+        sc = new Scanner(System.in);
+        out = System.out;
+    }
+
+    /**
+     * Shows Greetings Ui
+     */
+    public String showGreetings() {
+        String greetings = "_______________________________________\nHello! I'm Duke"
+                + "\nWhat can I do for you?\n_______________________________________";
+        return greetings;
+    }
+
+    /**
+     * Shows Bye Ui
+     * @return string that contains bye message
+     */
+    public static String showBye() {
+        return ("_______________________________________\nBye. Hope to see you again soon!\n"
+                + "_______________________________________\n");
+    }
+
+    /**
+     * Shows message when directory is created
+     */
+    public static void showDirectoryCreation() {
+        System.out.println("Creating new /data/ directory folder.");
+    }
+
+    /**
+     * Shows message when file is created
+     */
+    public static void showFileCreation() {
+        System.out.println("Creating new duke.txt file under /data/ directory folder.");
+    }
+
+    /**
+     * Shows successful load
+     */
+    public static void showLoadSuccessful() {
+        System.out.println("Loaded successfully!");
+    }
+
+    /**
+     * Shows successful save
+     */
+    public static void showSaveSuccessful() {
+        System.out.println("Saved successfully!");
+    }
+
+    /**
+     * Shows the tasks in taskList
+     * @param taskList the tasks to be shown
+     * @return string that contains tasks in list
+     */
+    public static String showTasksInList(TaskList taskList) {
+        StringBuilder stringBuilder = new StringBuilder("Here are the tasks in your list:\n");
+        for (int i = 0; i < taskList.taskListSize(); i++) {
+            stringBuilder.append(i + 1 + ". " + taskList.getTask(i).toString() + "\n");
+        }
+        return stringBuilder.toString();
+    }
+
+    /**
+     * Reads the command entered by user
+     * @return the command entered by user
+     */
+    public String readCommand() {
+        return sc.nextLine();
+    }
+
+    /**
+     * Shows the error message
+     * @param message the error message
+     * @return string that contains error message
+     */
+    public String showError(String message) {
+        return message;
+    }
+
+    /**
+     * Shows when task is added
+     * @param taskList the taskList that the task is added to
+     * @return string that contains task added
+     */
+    public String showTaskAdded(TaskList taskList) {
+        return ("Got it. I've added this task:")
+                + "\n" + (taskList.latestTask() + "\nNow you have " + taskList.taskListSize()
```

Comment:
I really liked there is a consistent breaking up of long lines into two or more lines across the entire project